Tests/scripts/test_playbooks.py

```diff
@@ -0,0 +1,208 @@
+from typing import Any
+
+import tenacity
+from jira import JIRAError, JIRA, Issue
+from jira.client import ResultList
+
+from junitparser import JUnitXml, TestSuite
+from tqdm import tqdm
+
+from Tests.scripts.jira_issues import generate_ticket_summary, generate_query
+from Tests.scripts.utils import logging_wrapper as logging
+
+TOTAL_HEADER = "Total"
+NOT_AVAILABLE = "N/A"
+TEST_SUITE_JIRA_HEADERS = ["Jira Ticket", "Jira Ticket Resolution"]
+TEST_SUITE_BASE_HEADERS = ["Playbook ID"]
+TEST_SUITE_FIXED_HEADERS = TEST_SUITE_BASE_HEADERS + TEST_SUITE_JIRA_HEADERS
+TEST_SUITE_DATA_CELL_HEADER = "S/F/E/T"
+NO_COLOR_ESCAPE_CHAR = "\033[0m"
+RED_COLOR = "\033[91m"
+GREEN_COLOR = "\033[92m"
+
+
+class TestSuiteDataCell:
+    def __init__(self, failures: int = 0, errors: int = 0, skipped: int = 0, tests: int = 0):
+        self.failures = failures
+        self.errors = errors
+        self.skipped = skipped
+        self.tests = tests
+
+    def __add__(self, other):
+        return TestSuiteDataCell(self.failures + other.failures, self.errors + other.errors, self.skipped + other.skipped,
+                                 self.tests + other.tests)
+
+    def __str__(self):
+        return f"{self.skipped}/{self.failures}/{self.errors}/{self.tests}"
+
+
+def calculate_test_playbooks_results(test_playbooks_result_files_list: list[Path]) -> tuple[dict[str, dict[str, Any]], set[str]]:
+    playbooks_results: dict[str, dict[str, Any]] = {}
+    server_versions = set()
+    for test_playbook_result_file in test_playbooks_result_files_list:
+        xml = JUnitXml.fromfile(test_playbook_result_file.as_posix())
+        for test_suite_item in xml.iterchildren(TestSuite):
+            properties = {prop.name: prop.value for prop in test_suite_item.properties()}
+            playbooks_result = playbooks_results.setdefault(properties["playbook_id"], {})
+            server_version = properties["server_version"]
+            server_versions.add(server_version)
+            playbooks_result[server_version] = test_suite_item
+    return playbooks_results, server_versions
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=1, min=2, max=10),
+    stop=tenacity.stop_after_attempt(5),
+    reraise=True,
+    retry=tenacity.retry_if_exception_type(JIRAError),
+    before_sleep=tenacity.before_sleep_log(logging.getLogger(), logging.DEBUG)
+)
+def search_ticket_in_jira(jira_server: JIRA, playbook_id: str) -> Issue | None:
+    jira_ticket_summary = generate_ticket_summary(playbook_id)
+    jql_query = generate_query(jira_ticket_summary)
+    search_issues: ResultList[Issue] = jira_server.search_issues(jql_query)  # type: ignore[assignment]
+    if search_issues:
+        playbook_id_lower = playbook_id.lower()
+        for issue in search_issues:
+            if playbook_id_lower in issue.get_field("summary").lower():
+                return issue
+        logging.debug(f"Failed to find a jira ticket for playbook id: {playbook_id}")
```

Comment:
Deserved to be an `info` level log, no?
Or you're searching for open issues only? (doesn't look like it)

Reply:
only searching for unresolved issues